src/App.tsx

```diff
@@ -1,24 +1,61 @@
 import React from 'react';
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
+import { Clock } from './components/Clock/Clock';
+
+interface State {
+  isClockVisible: boolean;
+  clockName: number;
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: Math.round(Math.random() * 1000),
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  renameClock = () => {
+    this.setState({ clockName: Math.round(Math.random() * 1000) });
+  };
+
+  render() {
+    const { isClockVisible } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <button
+          type="button"
+          onClick={this.hideClock}
+        >
+          Hide Clock
+        </button>
+        {' '}
+
+        <button
+          type="button"
+          onClick={this.showClock}
+        >
+          Show Clock
+        </button>
```

Comment:
Not necessary for approve, but if you want to improve your solution, you may change the disabled state of buttons if the clock is hidden/shown